third_party/terraform/data_sources/data_source_monitoring_service_app_engine.go

```diff
@@ -0,0 +1,34 @@
+package google
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceMonitoringServiceAppEngine() *schema.Resource {
+	aeSchema := map[string]*schema.Schema{
+		"module_id": {
+			Type:     schema.TypeString,
+			Required: true,
+			Description: `The ID of the App Engine module underlying this service. 
+Corresponds to the 'moduleId' resource label for a 'gae_app'
+monitored resource(see https://cloud.google.com/monitoring/api/resources#tag_gae_app)`,
+		},
+	}
+	filter := `app_engine.module_id="{{module_id}}"`
+	return dataSourceMonitoringServiceType(aeSchema, filter, dataSourceMonitoringServiceAppEngineRead)
+}
+
+func dataSourceMonitoringServiceAppEngineRead(res map[string]interface{}, d *schema.ResourceData, meta interface{}) error {
```

Comment:
This is being passed as the "flattenFunc" but it's calling d.Set and it's not actually making a Get call to a resource. I assumed that there was a syntax error earlier when the flatten method was called but the only thing returned was an error. To me a flatten method is a transform and a Read fetches a resource. This is more of a setter than either of those. Maybe rename to `dataSourceMonitoringServiceAppEngineStateSetter`?